docs/cli/tkgctl/bom.md

```diff
@@ -0,0 +1,124 @@
+
+* Tanzu CLI is decoupled from TKG BoM file (and TKR BoM file) and won't be bundled in CLI library.
+* Tanzu CLI is compiled with TKG Compatibility Image path( build time constant ). Tanzu CLI at runtime downloads the compatibility file and determines the TKG BoM image path from the compatibility file (version matching the management-cluster plugin version currently installed) and downloads the BoM files as well to the user's machine. [reference](../../../tkg/tkgconfigupdater/ensure.go)
+
+### Why do we need TKG Compatibility Image?
+
+If the BoM files are bundled with CLI, and when there is a CVE fix in any images/components that are part of TKG/TKR BoM file for an existing release, this would warrant a new CLI release.
+By using a TKG Compatibility Image, a new CLI release can be avoided. The release team can publish a new compatibility image(new Tag) with updated TKG BoM paths for a given management-cluster plugin version, and CLI would download the new Compatibility Image(with the latest Image Tag) and there by can download and use the updated TKG/TKR BoM files while creating a new management cluster.
+
+Sample compatibility file is as shown below:
+
+```yaml
+version: v2
+managementClusterPluginVersions:
+- version: v1.5.1
+  supportedTKGBomVersions:
+  - imagePath: tkg-bom
+    tag: v1.5.1
+- version: v1.5.0
+  supportedTKGBomVersions:
+  - imagePath: tkg-bom
+     tag: v1.5.0
+```
+
+Sample compatibility file after the updating the BoM version(new BoM which has images/components with CVE fixes for v1.5.0) is shown below. The new compatibility image would be pushed to image repository with updated tag( eg: **tkg/tkg-compatibility:v3**)
+
+```yaml
+version: v3
+managementClusterPluginVersions:
+  - version: v1.5.1
+    supportedTKGBomVersions:
+      - imagePath: tkg-bom
+        tag: v1.5.1
+  - version: v1.5.0
+    supportedTKGBomVersions:
+      - imagePath: tkg-bom
+          tag: v1.5.0-patch
+```
+
+Notes:
+
+* Once we create a management cluster it is running the TKR controller which reconciles all the supported TKR and updates TKR compatibility.
+* For workload cluster creation, CLI can download the necessary TKR to the user's local machine from the management cluster (existing ConfigMap on MC) and uses downloaded TKR to create a workload cluster.
+
+### Why do we need to download TKR BoM locally to the user's machine if not present locally?
+
+* This is required because cluster template creation logic is still running locally with the TKG library which internally uses YTT
+* To generate cluster templates using YTT overlays and easy implementation we are downloading the TKR BoM file locally before generating the cluster template
+* This allows users to read and understand the content of the TKR BoM file for debugging purpose
+
+## Updating TKG Compatibility Image Path into the TKG library
+
+1. Update `TKG_DEFAULT_IMAGE_REPOSITORY` and `TKG_DEFAULT_COMPATIBILITY_IMAGE_PATH` variable inside [Makefile](../../../Makefile)
```

Comment:
Should we update the variables in the `Makefile` directly, or simply set them for the `Makefile` to read?

Reply:
These steps are for making permanent changes.  If we want to test locally, we can either set variables values in Makefile for `TKG_DEFAULT_IMAGE_REPOSITORY` and `TKG_DEFAULT_COMPATIBILITY_IMAGE_PATH` or export the env variables for the makefile to read them. Later ignore the step 3) and instead can use `make build-install-cli-local` to build and test locally.